### PR TITLE
[Feature] Add function upsample_like and its unit test

### PR DESCRIPTION
**Pull request by AronLin**
## Motivation

The preparation work for Panoptic Segmentation, used in semantic heads.
This function upsamples the logits to the shape of the target.
The input logits can be 3D or 4D. The target can be a Tensor or a `np.ndarray` with the shape (..., target_h, target_w).

## BC-breaking (Optional)

No.

### CODE REVIEW

File: mmdet/models/utils/misc.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from torch.nn import functional as F
+
+
+def upsample_like(logits, target, mode='bilinear', align_corners=False):
+    """Up-sample the logits to the shape of target.
```
Comment (ZwwWayne) [inline mmdet/models/utils/misc.py:5]: Upsample

Comment (AronLin) [reply mmdet/models/utils/misc.py]: Done.

**Review by ZwwWayne — APPROVED** [review]
